### PR TITLE
Multiple Bugfixes mostly based on validation layer errors

### PR DESCRIPTION
**Pull request by jhmueller-huawei**
- SpecularReflectionsFeatureProcessor.cpp: remove asserts that unnecessarily trigger when UpdatePasses is called before AttachmentBindings are created.
- MultiDeviceObject derived classes: Clear map of DeviceObjects on Shutdown and do not forward the Shutdown call as the DeviceObjects may still be in use.
- Fence: only use timeline semaphores in Vulkan when waiting for a Fence on the GPU. Fixes the bug that you cannot immediately free a timeline semaphore when it is signaled as is done with the Fence in the AsyncUploadQueue.
- Vulkan FenceBase/BinaryFence/TimelineSemaphoreFence: fixed wrong Shutdown chain
- VkAccelerationStructure: fix leak by implementing a RayTracingAccelerationStructure class with a Shutdown method that properly releases the Vulkan object. To keep it around long enough it is referenced in the Buffer class that stores the corresponding acceleration structure.
- Vulkan RayTracingAccelerationStructure: keep a vector of references to BLAS buffers to keep them alive as long as the TLAS exists.
- Vulkan Device: m_swapChaiSemaphoreAllocator name spelling fixed to m_swapChainSemaphoreAllocator (chaiN).
- Vulkan Device: properly call m_swapChainSemaphoreAllocator's Collect() and Shutdown() methods to stop leaking semaphores.
- Vulkan Device: fixed release order in Device::PreShutdown.
- Vulkan RayTracingPipelineState: m_pipeline was leaking.
- Vulkan Semaphore: removed unnecessary SignalEvent call.
- Vulkan SwapChain: removed unnecessary SignalEvent call that triggered an assert.
- Vulkan SwapChain: wait for the graphics queue only instead of the whole device which causes an issue if another thread works on the AsyncUploadQueue at the same time.

## What does this PR do?

I got a lot of validation layer errors in the atom-sampleviewer especially when closing it on shutdown. This PR fixes many (but not all) of these.

## How was this PR tested?

Run the atom-sampleviewer with `--rhi-device-validation enable` and the Vulkan backend (default on Linux).

### CODE REVIEW

File: Gems/Atom/RHI/Code/Include/Atom/RHI/DeviceFence.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace AZ::RHI
         virtual ~DeviceFence() = 0;
 
         /// Initializes the fence using the provided device and initial state.
```
Comment (martinwinter-huawei) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/DeviceFence.h:27]: Could you add the description of the new parameter to the comment.
Also, not sure if `usedForWaiting` is the best name, maybe `usedForWaitingOnGPU` or something along those lines to be more specific?

File: Gems/Atom/RHI/Code/Include/Atom/RHI/Fence.h [hunk 2]
```diff
@@ -27,7 +27,7 @@ namespace AZ::RHI
         //! Initializes the multi-device fence using the provided deviceMask.
         //! It creates on device-specific fence for each bit set in the deviceMask and
         //! passes on the initial FenceState to each DeviceFence
-        ResultCode Init(MultiDevice::DeviceMask deviceMask, FenceState initialState);
+        ResultCode Init(MultiDevice::DeviceMask deviceMask, FenceState initialState, bool usedForWaiting = false);
```
Comment (martinwinter-huawei) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/Fence.h:30]: Same here

**Review by martinwinter-huawei — APPROVED** [review]
Nice job tracking all those down 👍

File: Gems/Atom/RHI/Code/Source/RHI/Buffer.cpp [hunk 3]
```diff
@@ -45,11 +45,6 @@ namespace AZ::RHI
 
     void Buffer::Shutdown()
     {
-        IterateObjects<DeviceBuffer>([]([[maybe_unused]] auto deviceIndex, auto deviceBuffer)
```
Comment (moudgils) [inline Gems/Atom/RHI/Code/Source/RHI/Buffer.cpp:48]: Are these Shutdown calls moved somewhere else?

Comment (jhmueller-huawei) [reply Gems/Atom/RHI/Code/Source/RHI/Buffer.cpp:48]: They are called when the DeviceObjects are freed which is the case if they are not used elsewhere anymore by clearing the map here. The reason for this change is precisely that they can still be used in a command buffer in flight and will be released later. We tested this and everything works fine. The only explicit call outside of destruction we found is in the RPI StreamingImage and that still works as expected.

**Pull request merged**